### PR TITLE
updated unit tests & wrote some end-to-end tests

### PR DESCRIPTION
**Pull request by alizehrehman**

### CODE REVIEW

**Comment by alizehrehman** [comment]
- Updated jest config files to give tests enough time to render and run
- Updated jest config to be verbose
- Changed dummyData to be easier to use in tests
- Added unit tests
- Added end-to-end test to ensure the up/down scroll is working as expected
- Updated some components to be more "testable"

**Comment by alizehrehman** [comment]
@matthewwrobel @jabrake Please review my PR tomorrow. Thank you!

File: client/product_details/Testing/unitTests.test.js [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+      selectedStyle={data.fiveStyles.results[0]}
+      indexSelectedStyle={0}
+      handleStyleSelection={handleStyleSelection}
+    />, container);
+  });
+  
+  var thumbnailList = document.getElementById('thumbnailList');
+  var numberOfThumbnails = thumbnailList.childElementCount;
+
+  expect(numberOfThumbnails).toBe(10);
```
Comment (jabrake) [inline client/product_details/Testing/unitTests.test.js:107]: Should this be `expect(numberOfThumbnails).toBe(7)`? Or is it checking that there are 10 children, but only 7 are rendering on the screen?

File: client/product_details/Testing/endToEndTests.test.js [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    const lastThumbnail = await page.$$('#7');
+    const firstThumbnail = await page.$$('#0');
+
+    expect(thumbnails.length).toBe(7);
+    expect(lastThumbnail).toBeTruthy();
+    expect(firstThumbnail).toBeNull();
+  })
+  it('should scroll correctly after seven up-clicks', async () => {
+    await page.goto(testUrl);
+    await page.click('#upScroll-container');
```
Comment (jabrake) [inline client/product_details/Testing/endToEndTests.test.js:81]: Can lines 81-127 be simplified into some kind of for loop?

**Review by jabrake — APPROVED** [review]
Looks great! Just a few comments but ready to merge.

File: client/product_details/Testing/unitTests.test.js (hunk 1 above)
Comment (alizehrehman) [reply client/product_details/Testing/unitTests.test.js:107]: @jabrake I actually have two icons and a break in that component as well as the images. Unfortunately I couldn't figure out how to isolate the selection by child and image so I thought I'd get around that by just check that it has the 3 children plus the 7 images.

File: client/product_details/Testing/endToEndTests.test.js (hunk 2 above)
Comment (alizehrehman) [reply client/product_details/Testing/endToEndTests.test.js:81]: That sounds like a great idea! Might be able to make it a separate function? I will try it out later when I'm trying to use these tests to debug the component - some strange things are happening to the indices!

**Pull request merged**